### ISSUE TITLE
Mermaid docker compose and Dockerfile build fails on RUN apk add --no-cache git~=2.43.4

### ISSUE THREAD

**Issue by okoeroo** [mermaid-js/mermaid#6154]
### Description

Error is the following:
 => ERROR [mermaid 3/3] RUN apk add --no-cache git~=2.43.4     && git config --add --system safe.directory /mermaid

Solved by removing the git version specification.

### Steps to reproduce

git clone main
docker compose up

failure.

### Screenshots

 => => extracting sha256:db3e2f2b60543c32c84a427cc2129966358695d5b298456169c12df7ee2fd1e0                                                                                                                        0.0s
 => [mermaid 2/3] RUN corepack enable     && corepack enable pnpm                                                                                                                                                1.2s
 => ERROR [mermaid 3/3] RUN apk add --no-cache git~=2.43.4     && git config --add --system safe.directory /mermaid                                                                                              1.1s
------
 > [mermaid 3/3] RUN apk add --no-cache git~=2.43.4     && git config --add --system safe.directory /mermaid:
0.240 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/main/x86_64/APKINDEX.tar.gz
0.637 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/community/x86_64/APKINDEX.tar.gz
1.019 ERROR: unable to select packages:
1.020   git-2.43.5-r0:
[+] Running 0/1s: world[git~2.43.4]
 ⠦ Service mermaid  Building                                                                                                                                                                                    18.6s
failed to solve: process "/bin/sh -c apk add --no-cache git~=2.43.4     && git config --add --system safe.directory /mermaid" did not complete successfully: exit code: 1

### Setup

- Mermaid version: main (28 dec 2024)

### Suggested Solutions

Remove git version in apk add

**Comment by ShortArrow** [comment]
# Quickfix when run docs dev server on local docker

```bash
git clone https://github.com/mermaid-js/mermaid  -b develop
sed -i 's/~=2.43.4//' ./Dockerfile
sed -i -e 's/pnpm run --filter mermaid docs:dev:docker/pnpm install \&\& pnpm run --filter mermaid docs:dev:docker/' ./run
./run docs:dev
```

**Comment by mekasone** [comment]
Just delete the following from the Dokcerfilye (make sure to copy the original before modifying, just in case you need go back to the original):  
~=2.43.4

**Comment by hmzq-dev** [comment]
@okoeroo can I add a pull request to remove the git version specification from Dockerfile?

**Comment by ShortArrow** [comment]
@mekasone Yes, `./run pnpm install` is available. I realized it thanks to you.

https://github.com/mermaid-js/mermaid/blob/074701e31695cf3478043e9630ca196484e1d600/run#L26
